public/docs/ts/latest/cookbook/aot-compiler.jade

```diff
@@ -0,0 +1,177 @@
+
+    The downside to JIT compilation is performance. Views take longer to render because of the compilation step. It also forces us to download the Angular compiler along with our application code since we will need the compiler at runtime.  
+    
+    In this cookbook we discuss how we can improve performance by doing compilation at build time instead. We call this process Ahead of Time compilation (AOT). 
+<a id="toc"></a>
+:marked
+   ## Table of contents   
+
+   [Ahead of Time vs Just in Time](#aot-jit)
+
+   [Compilation](#compilation)
+
+   [Tree Shaking](#tree-shaking)
+
+   [Bootstrapping](#bootstrapping)
+
+.l-main-section
+<a id="aot-jit"></a>
+:marked
+    ## Ahead of Time (AOT) vs Just in Time (JIT)
+
+    AOT and JIT compilation use the same exact compiler. The difference is, AOT happens at build time, JIT happens at runtime.  
+
+    ### Why do AOT compilation?
+
+    The performance improvement from doing AOT compilation can be significant for two reasons:
+
+    *Faster rendering*
+
+    When doing AOT compilation the browser downloads a pre-compiled version of the application. This means the browser can start rendering the application immediately, without delays from doing JIT compilation first.
+
+    The compiler will also inline external files like html templates and css files, which means we no longer need separate ajax requests to load external files at run time.
+
+    *Smaller Angular framework download size*
+
+    Compilation at build time means we no longer have to download the compiler with our application. The size of the compiler is significant, so omitting it will automatically reduce the download size of Angular.
+
+.l-main-section
+<a id="compilation"></a>
+:marked
+    ## Compilation
+
+    ### Prepare for offline compilation
+
+    Before we can AOT compile our application we have to install a few npm dependencies by running: 
+
+    `npm install @angular/compiler-cli typescript@2.0.0 @angular/platform-server @angular/compiler`
+    
+    When doing AOT compilation we no longer use the regular TypeScript compiler (`tsc`) directly. Instead we use the `ngc` compiler provided by the `@angular/compiler-cli` npm package. 
+    
+    `ngc` is a drop-in replacement for `tsc`.
+    
+    We configure the `ngc` compiler using a tsconfig.json file the same way we would if we were using `tsc`.  
+
+    ### Compiling the application
+
+    We trigger AOT compilation via the command line using the previously installed `ngc` compiler by executing:
+
+    `node_modules/.bin/ngc -p [path to project]`
+
+    `ngc` expects the `-p` switch to point to a tsconfig.json file or a folder containing a tsconfig.json file.
+
+    The first thing we notice after running `ngc` is a series of generated files with an ngfactory suffix.
```

Comment:
Before we tell them to run `ngc` it may be good to have them set the gen folder so the files do not go side by side.